peer.go

```diff
@@ -200,9 +198,9 @@ func (p *peer) downloadPiece(piece *Piece) ([]byte, error) {
 	return res, nil
 }
 
-// Download creates a new peer that downloads pieces from a file
-func Download(handshake []byte, address string, pieces chan *Piece, results chan<- *Result) {
-	peer, err := new(handshake, address)
+// DownloadPieces creates a new peer that downloads pieces from a file
+func DownloadPieces(handshake []byte, address string, pieces chan *Piece, results chan<- *Result) {
+	peer, err := newPeer(handshake, address)
```

Comment:
this should probably be a method on peer, right?

Reply:
It might, but the creation of the peer would have to be moved higher up. In another PR?

Reply:
sure.